### ISSUE TITLE
integration flake: test-update-storage-objects.sh Unable to perform initial IP allocation check

### ISSUE THREAD

**Issue by gmarek** [kubernetes/kubernetes#19467]
https://console.developers.google.com/storage/browser/kubernetes-jenkins/pr-logs/a8d913bf8bbdfce90d4d2ee10f1a4c527f7aa14f/kubernetes-pull-test-unit-integration/8699/

cc @lavalamp @krousey @ixdy

**Comment by gmarek** [comment]
https://console.developers.google.com/storage/browser/kubernetes-jenkins/pr-logs/e7e8c5af8421c3f1fdb79e2159a3483615d56153/kubernetes-pull-test-unit-integration/8543/

**Comment by ixdy** [comment]
This is `hack/test-update-storage-objects.sh`.

cc @nikhiljindal

**Comment by lavalamp** [comment]
The error is in recovering the service IP allocations:

```
F0108 19:58:11.963876   32590 controller.go:84] Unable to perform initial IP allocation check: unable to persist the updated service IP allocations: serviceipallocations "" cannot be updated: the provided resource version does not match
```

This seems like some sort of legit problem in https://github.com/kubernetes/kubernetes/blob/c059dfdb9bf05fc3dec96243b6101cf84f2db4ef/pkg/registry/service/allocator/etcd/etcd.go#L198 or the arguments that it's called with.

@thockin Can you delegate to the current owner of this code?

**Comment by thockin** [comment]
@smarterclayton

**Comment by brendandburns** [comment]
I took a stab at both getting more info (and possibly fixing the flake)

**Comment by smarterclayton** [comment]
This is a dupe of #19848 and I already have a fix https://github.com/openshift/origin/pull/6737 which is waiting on https://github.com/kubernetes/kubernetes/pull/16067

**Comment by thockin** [comment]
Should we close?

**Comment by smarterclayton** [comment]
Fixing now.

**Comment by smarterclayton** [comment]
#20007

**Comment by lavalamp** [comment]
Should be fixed now that #20007 has merged.
